File: 724.ts
function pivotIndex(nums: number[]): number {
  let sum = 0;
  let prefixSums = [0];

  for (let num of nums) {
    sum += num;
    prefixSums.push(sum);
  }

  for (let i = 0; i < nums.length; i++) {
    let sumBefore = prefixSums[i];
    let sumAfter = prefixSums[prefixSums.length - 1] - prefixSums[i + 1];

    if (sumBefore == sumAfter) {
      return i;
    }
  }

  return -1;
}

const arr = [1, 7, 3, 6, 5, 6];
console.log(pivotIndex(arr));

File: 724.py
from typing import List


class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        prefix_sums = [0]
        sum = 0

        for num in nums:
            sum += num
            prefix_sums.append(sum)

        for i in range(len(nums)):
            if prefix_sums[i] - prefix_sums[0] == prefix_sums[-1] - prefix_sums[i+1]:
                return i

        return -1
    

sol = Solution()
sol.pivotIndex([1,7,3,6,5,6])